hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSDirectory.java

```diff
@@ -1091,6 +1091,24 @@ static void unprotectedUpdateCount(INodesInPath inodesInPath,
     }
   }
 
+  /**
+   * check that all parent directories have quotas set.
+   */
+  static boolean verifyIsQuota(INodesInPath iip, int pos) {
+    for (int i = (Math.min(pos, iip.length()) - 1); i > 0; i--) {
+      INode currNode = iip.getINode(i);
+      if (currNode == null) {
+        continue;
+      }
+      if (currNode.isDirectory()) {
+        if (currNode.isQuotaSet()) {
```

Comment:
if the bottom most directory has quota set, does it imply all the ancestor directories have quota set?